pycbc/inference/models/relbin.py

```diff
@@ -440,26 +484,38 @@ def _loglr(self):
         self._current_wf_parts = {}
         for ifo in self.data:
 
-            det = self.det[ifo]
             freqs = self.fedges[ifo]
             sdat = self.sdat[ifo]
             h00 = self.h00_sparse[ifo]
             end_time = self.end_time[ifo]
             times = self.antenna_time[ifo]
 
-            # project waveform to detector frame
-            hp, hc = wfs[ifo]
-            fp, fc = det.antenna_pattern(p["ra"], p["dec"],
-                                         p["polarization"], times)
-            dt = det.time_delay_from_earth_center(p["ra"], p["dec"], times)
-            dtc = p["tc"] + dt - end_time
-
-            hdp, hhp = self.lik(freqs, fp, fc, dtc,
-                                hp, hc, h00,
-                                sdat['a0'], sdat['a1'],
-                                sdat['b0'], sdat['b1'])
-
-            self._current_wf_parts[ifo] = (fp, fc, dtc, hp, hc, h00)
+            # project waveform to detector frame if waveform does not deal
+            # with detector response. Otherwise, skip detector response.
+
+            if self.det_response:
+                fp = 1
+                dtc = -end_time
+                print(wfs[ifo])
+                hp = wfs[ifo]
+                hdp, hhp = self.lik(freqs, fp, dtc,
+                                    hp, h00,
+                                    sdat['a0'], sdat['a1'],
+                                    sdat['b0'], sdat['b1'])
+                self._current_wf_parts[ifo] = (fp, dtc, hp, h00)
```

Comment:
Remove this line. It is needed for multi-signal support, but we can't handle that case at the moment when using this kind of model.

Reply:
Removed